src/apps/persistence/docker_registries.clj

```diff
@@ -13,6 +14,15 @@
   (first (select docker-registries
                  (where {:name name}))))
 
+(defn get-registry-from-image
+  [image-name]
+  (let [parts (string/split image-name #"/")]
+    (loop [n (count parts)]
+      (when-not (= n 0)
+        (if-let [reg (get-registry (string/join "/" (take n parts)))]
```

Comment:
`get-registry` being inside this loop does mean this can do up to `(count parts)` queries against the DB. I'm wondering if that's worth fixing or not; I guess it would most affect deeply-nested image names that don't require auth (e.g. something like `foo.bar.baz/project/subproject/subsubproject/image` would query five times)